### ISSUE TITLE
Maybe disable gzip in nginx config only when there is an ETag header?

### ISSUE THREAD

**Issue by benediktg** [nextcloud/documentation#325]
I have found this page from the nginx docs: https://nginx.org/en/docs/http/ngx_http_gzip_module.html

It says about the `gzip_proxied` statement:

> Enables or disables gzipping of responses for proxied requests depending on the request and response. The fact that the request is proxied is determined by the presence of the “Via” request header field. The directive accepts multiple parameters: 
    […]
    `no_etag`
    enables compression if a response header does not include the “ETag” field; 

Could it make sense to enable gzip but only disable it for responses containing the ETag header (in the [nginx config recommendation](https://github.com/nextcloud/documentation/blob/master/admin_manual/installation/nginx_nextcloud_9x.rst))?

**Comment by MorrisJobke** [comment]
@LukasReschke @rullzer The etag header issues where to only problem here, right?

@josh4trunks Does it make sense to use this parameter?

**Comment by josh4trunks** [comment]
I'm not very informed on this subject. My production setup has gzip disabled with nginx, and instead text/application files are gzipped upstream by my varnish proxy.

Is there a good test URL to see when gzip causes issues in nextcloud? Maybe @benediktg you could verify gzip causes issues normally, but this parameter fixes it.

**Comment by benediktg** [comment]
My current gzip configuration is
```nginx
gzip on;
gzip_comp_level 5;
gzip_min_length 256;
gzip_proxied expired no-cache no-store private no_last_modified no_etag auth;
gzip_types
    application/atom+xml
    application/javascript
    application/json
    application/ld+json
    application/manifest+json
    application/rss+xml
    application/vnd.geo+json
    application/vnd.ms-fontobject
    application/x-font-ttf
    application/x-web-app-manifest+json
    application/xhtml+xml
    application/xml
    font/opentype
    image/bmp
    image/svg+xml
    image/x-icon
    text/cache-manifest
    text/css
    text/plain
    text/vcard
    text/vnd.rim.location.xloc
    text/vtt
    text/x-component
    text/x-cross-domain-policy;
```
and as far as I can say everything seems to still work as expected.

**Comment by michaelletzgus** [comment]
I have compression completely enabled without any problems so far.
The weak ETag seem to work quite well:
 
```Nginx
   gzip on;
        gzip_vary on;
        gzip_comp_level 5;
        gzip_min_length 128;
        gzip_buffers 32 16k;
        gzip_proxied any;
        gzip_types application/javascript application/x-javascript text/javascript application/json text/xml application/xml text/css text/vcard image/svg+xml image/x-icon;

```

**Comment by joshtrichards** [comment]
Can be closed, no?

This was addressed long ago in #386 (which was triggered by this very issue).
